File: abc/abc185/abc185c.c
#include <stdio.h>

long long gcd(long long x, long long y) {
    long long t;

    if (x < y) {
        t = x;
        x = y;
        y = t;
    }

  	while (y > 0) {
        t = x;
        x = y;
        y = t % y;
    }
    return x;
}

int main() {
    int L;
    long long d = 39916800; // 11!
    long long result = 1;
    long long t;

    scanf("%d", &L);

    for (int i = 0; i < 11; i++) {
        t = gcd(L - 1 - i, d);
        result *= (L - 1 - i) / t;
        d /= t;
    }

    printf("%lld", result);
}

File: abc/abc185/abc185c.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

func gcd(x, y int) int {
	if x < y {
		x, y = y, x
	}
	for y > 0 {
		x, y = y, x%y
	}
	return x
}

func main() {
	defer flush()

	L := readInt()

	d := 1
	for i := 0; i < 11; i++ {
		d *= i + 1
	}

	result := 1
	for i := 0; i < 11; i++ {
		t := gcd(L-1-i, d)
		result *= (L - 1 - i) / t
		d /= t
	}

	println(result)
}

const (
	ioBufferSize = 1 * 1024 * 1024 // 1 MB
)

var stdinScanner = func() *bufio.Scanner {
	result := bufio.NewScanner(os.Stdin)
	result.Buffer(make([]byte, ioBufferSize), ioBufferSize)
	result.Split(bufio.ScanWords)
	return result
}()

func readString() string {
	stdinScanner.Scan()
	return stdinScanner.Text()
}

func readInt() int {
	result, err := strconv.Atoi(readString())
	if err != nil {
		panic(err)
	}
	return result
}

var stdoutWriter = bufio.NewWriter(os.Stdout)

func flush() {
	stdoutWriter.Flush()
}

func println(args ...interface{}) (int, error) {
	return fmt.Fprintln(stdoutWriter, args...)
}
